core/src/test/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateSpec.scala

```diff
@@ -113,6 +114,40 @@ abstract class JdbcDurableStateSpec(config: Config, schemaType: SchemaType) exte
         }
       }
     }
+    "fail to delete old object revision" in {
+      val f = for {
+        n <- stateStoreString.upsertObject("p987", 1, "a valid string", "t123")
+        _ = n shouldBe pekko.Done
+        g <- stateStoreString.getObject("p987")
+        _ = g.value shouldBe Some("a valid string")
+        u <- stateStoreString.upsertObject("p987", 2, "updated valid string", "t123")
+        _ = u shouldBe pekko.Done
+        d <- stateStoreString.deleteObject("p987", 1)
+      } yield d
+      whenReady(f.failed) { e =>
+        e shouldBe a[DurableStateStoreException]
```

Comment:
WIP here - I still need to rewrite this test to make the correct assertions depending on whether pekko-persistence 1.0.x is used or 1.1.x is used.

Reply:
test rewritten but needs a bit more manual testing from me